pre/remap_restart/remap_utils.py

```diff
@@ -6,9 +6,268 @@
 from collections import OrderedDict
 import shutil
 import questionary
+import glob
+import shlex
+import netCDF4 as nc
 
-def config_to_yaml(config, yaml_file):
-   if os.path.exists(yaml_file) :
+# shared global variables
+#
+# define "bc_base", "choices", "message" strings, and "validate" lists that are used multiple times
+#   (and related definitions, even if they are used just once).
+
+bc_base = "/discover/nobackup/projects/gmao/bcs_shared/fvInput/ExtData/esm/tiles"
```

Comment:
@biljanaorescanin, @weiyuan-jiang :  `bc_base` here is hardwired for Discover.  Does this mean that the tool won't work at NAS?  I didn't see an opportunity for users to overwrite `bc_base` (but might have missed it).
I suppose we will need to keep a copy of the official bcs dir at NAS if we're asking users to run off the official bcs (in progress PRs for gcm_setup/fv_setup ).    Then we may as well point remap_restarts to the corresponding NAS directory.  It's up to Scott to decide if this functionality has to be present before he agrees to merge this PR.  I'm ok either way, but we'll definitely need to work on this sooner or later.

Reply:
It does not! I don't think there is any reason it couldn't work at NAS if we had all the relevant BCs there (which we probably do though not the new BCs yet).

We'd also of course have to fix up all the bits about "what node type" do you want to run and sbatch v qsub, etc.

Reply:
So far it won't be working on NAS. But After NAS is ready for the new bcs, I will change this [line ](https://github.com/GEOS-ESM/GEOS_Util/blob/e84dddee8fbad8b263cfbd1169893d82ffb8b670/pre/remap_restart/remap_utils.py#L481)

@gmao-rreichle

Reply:
@weiyuan-jiang, thanks for the changes.  Once we know bc_base for NAS we can add it.  At that point, the tool will work at NCCS and NAS.  However, we still need the functionality for users to supply their own bc_base value.  Not all users are at NCCS or NAS.  Such users would have to edit `remap_utils.py` to get the tool to work for them, which isn't something we should ask of them.  The solution would be to include a question for bc_base in the questionary.  We can detect if we're at NCCS or NAS, and in this case we can skip the question and use the paths that are hardwired into `remap_utils.py`.  But if we're not at NCCS or NAS, we need to ask the user for bc_base.  
Since this functionality is needed anyway, we should probably do it now, because it will also make the code work at NAS immediately.  A NAS user just has to copy the (revised layout) bcs that they're using from NCCS into their own NAS directory temporarily.  Not great, but I'm still scarred from having to wait for the NCCS bcs_shared project directory for 12+ months.

Reply:
The question for bc_base is added. Also a 'customized' option is added for users' own base,